launcher/src/main/scripts/bin/launcher.py

```diff
@@ -69,16 +69,20 @@ def try_lock(f):
         return False
 
 
-def open_read_write(f, mode):
+def open_pidfile(f, mode):
     """Open file in read/write mode (without truncating it)"""
-    return os.fdopen(os.open(f, O_RDWR | O_CREAT, mode), 'r+')
+    fd = os.open(f, O_RDWR | O_CREAT, mode)
+    if hasattr(os, 'set_inheritable'):
```

Comment:
I think we can remove this same logic from the `start` method, now that we are setting it here.

Reply:
ah, good point, didn't realize this has already been dealt with elsewhere...
updating the PR to remove the same logic from the `start()` handler.